vault/resource_database_secret_backend_static_role.go

```diff
@@ -0,0 +1,210 @@
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	databaseSecretBackendStaticRoleBackendFromPathRegex = regexp.MustCompile("^(.+)/static-roles/.+$")
+	databaseSecretBackendStaticRoleNameFromPathRegex    = regexp.MustCompile("^.+/static-roles/(.+$)")
+)
+
+func databaseSecretBackendStaticRoleResource() *schema.Resource {
+	return &schema.Resource{
+		Create: databaseSecretBackendStaticRoleWrite,
+		Read:   databaseSecretBackendStaticRoleRead,
+		Update: databaseSecretBackendStaticRoleWrite,
+		Delete: databaseSecretBackendStaticRoleDelete,
+		Exists: databaseSecretBackendStaticRoleExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Unique name for the static role.",
+			},
+			"backend": {
```

Comment:
I've noticed that some folks name this "path", make it optional, and fall back to the secret engine name if it's not included (in this case, it would be "database" which is pretty generic.) I'm not dedicated to naming or handling it any one way. I was just curious, do you think we should do that here, or does that not make as much sense in this case?

Reply:
I tried to stay consistent with already existing [`vault_database_secret_backend_connection`](https://www.terraform.io/docs/providers/vault/r/database_secret_backend_role.html) resource, but we could change this for this. Or we could leave it as is and change this in another pull request in both db role resources. WDYT?